deegree-datastores/deegree-featurestores/deegree-featurestore-commons/src/main/java/org/deegree/feature/persistence/FeatureStoreManager.java

```diff
@@ -53,43 +50,58 @@ Occam Labs UG (haftungsbeschränkt)
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
 /**
  * Responsible for finding feature store resources.
- * 
+ *
  * @author <a href="mailto:schmitz@occamlabs.de">Andreas Schmitz</a>
  * @author last edited by: $Author: stranger $
- * 
  * @version $Revision: $, $Date: $
  */
 public class FeatureStoreManager extends DefaultResourceManager<FeatureStore> {
 
     private static Logger LOG = LoggerFactory.getLogger( FeatureStoreManager.class );
 
-    private static final String BBOX_CACHE_FILE = "bbox_cache.properties";
+    private static final String BBOX_CACHE_FILE = "bbox_cache_%s.properties";
 
-    private BBoxPropertiesCache bboxCache;
+    private final Map<String, BBoxPropertiesCache> bboxCaches = new HashMap<>();
+
+    private Workspace workspace;
 
     public FeatureStoreManager() {
-        super( new DefaultResourceManagerMetadata<FeatureStore>( FeatureStoreProvider.class, "feature stores",
-                                                                 "datasources/feature" ) );
+        super( new DefaultResourceManagerMetadata<>( FeatureStoreProvider.class, "feature stores",
+                                                     "datasources/feature" ) );
+    }
+
+    @Override public void startup( Workspace workspace ) {
+        this.workspace = workspace;
+        super.startup( workspace );
+    }
+
+    public BBoxCache getBBoxCache( String featureStoreId ) {
+        if ( !bboxCaches.containsValue( featureStoreId ) ) {
+            BBoxPropertiesCache bBoxCache = getOrCreateBBoxCache( featureStoreId );
+            bboxCaches.put( featureStoreId, bBoxCache );
+        }
+        return bboxCaches.get( featureStoreId );
     }
 
-    @Override
-    public void startup( Workspace workspace ) {
+    private BBoxPropertiesCache getOrCreateBBoxCache( String featureStoreId ) {
         try {
             if ( workspace instanceof DefaultWorkspace ) {
                 File dir = new File( ( (DefaultWorkspace) workspace ).getLocation(), getMetadata().getWorkspacePath() );
-                bboxCache = new BBoxPropertiesCache( new File( dir, BBOX_CACHE_FILE ) );
+                File propsFile = new File( dir, String.format( BBOX_CACHE_FILE, featureStoreId ) );
+                return new BBoxPropertiesCache( propsFile );
             }
-            // else?
         } catch ( IOException e ) {
-            LOG.error( "Unable to initialize global envelope cache: " + e.getMessage(), e );
+            LOG.error( "Unable to initialize envelope cache for feature store with id " + featureStoreId + ": "
+                       + e.getMessage(), e );
         }
-        super.startup( workspace );
-    }
-
-    public BBoxCache getBBoxCache() {
```

Comment:
Should this method be deleted, as it is now breaking implementations which reley on it.